frontend: replace vite-plugin-vue2-svg with vite-plugin-svg4vue

BacLuc opened the pull request:
Because vite-plugin-vue2-svg does not support vue3. Only worked when converting vite.config.js to
an esm file by renaming it to .mjs.
vite-plugin-svg4vue is not the most popular package for this, this would be https://www.npmjs.com/package/vite-svg-loader, but that does not support vue2.

Issue: #5121

We can also wait to merge this short before we want to upgrade to vue3.

Also fixes https://github.com/ecamp/ecamp3/security/dependabot/178

BacLuc committed:
frontend: replace vite-plugin-vue2-svg with vite-plugin-svg4vue

Because vite-plugin-vue2-svg does not support vue3.
Only worked when converting vite.config.js to
an esm file by renaming it to .mjs.
vite-plugin-svg4vue is not the most popular package
for this, this would be https://www.npmjs.com/package/vite-svg-loader,
but that does not support vue2.

Issue: #5121

We can also wait to merge this short before we want
to upgrade to vue3.

Also fixes https://github.com/ecamp/ecamp3/security/dependabot/178

usu reviewed (changes requested):
Seems not to work on deployment

![image](https://github.com/ecamp/ecamp3/assets/449555/0fdd62c3-2fa5-49cd-9119-5141d9dd88ed)

BacLuc committed:
fixup! frontend: replace vite-plugin-vue2-svg with vite-plugin-svg4vue

use v-html, because outputting the raw svg code into the element
does not work in production.

BacLuc commented:
> Seems not to work on deployment
> 
> ![image](https://private-user-images.githubusercontent.com/449555/333775642-0fdd62c3-2fa5-49cd-9119-5141d9dd88ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MzYxMzIsIm5iZiI6MTcxNjczNTgzMiwicGF0aCI6Ii80NDk1NTUvMzMzNzc1NjQyLTBmZGQ2MmMzLTJmYTUtNDljZC05MTE5LTUxNDFkOWRkODhlZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNlQxNTAzNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMjU1ZTI3YjcwMGM1NmJlZTQzNmI3YmExMGFmZTQ4MTgwZDgwZDcwNDQ4MWEyZDU0NmEzMjE2MzhiYmNmZGU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.t3fw9k3mwLTBpZJPYFDV9f11M0ZbPbZCHUzRMG3KpAQ)

I hoped we would get away without v-html.
It seems not

fixed

usu commented:
> > Seems not to work on deployment
> > ![image](https://private-user-images.githubusercontent.com/449555/333775642-0fdd62c3-2fa5-49cd-9119-5141d9dd88ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MzYxMzIsIm5iZiI6MTcxNjczNTgzMiwicGF0aCI6Ii80NDk1NTUvMzMzNzc1NjQyLTBmZGQ2MmMzLTJmYTUtNDljZC05MTE5LTUxNDFkOWRkODhlZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNlQxNTAzNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMjU1ZTI3YjcwMGM1NmJlZTQzNmI3YmExMGFmZTQ4MTgwZDgwZDcwNDQ4MWEyZDU0NmEzMjE2MzhiYmNmZGU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.t3fw9k3mwLTBpZJPYFDV9f11M0ZbPbZCHUzRMG3KpAQ)
> 
> I hoped we would get away without v-html. It seems not
> 
> fixed

Did the `?component` method not work? Or why do we need to use `?raw` and `v-html`?

usu commented on frontend/vite.config.mjs:
```diff
@@ -17,7 +17,9 @@ const plugins = [
   }),
-  createSvgPlugin(),
+  svg4VuePlugin({
+      skipSvgo: true
```
- What does svgo do? Why do we want to disable it?
- I think the option key should be `svgoConfig`, at least according to the [docs](https://github.com/yisibell/vite-plugin-svg4vue?tab=readme-ov-file#options).

manuelmeister replied:
It minimizes the svgs. I'm not sure this is the way to go

manuelmeister reviewed (changes requested):
Why do we want to use this plugin when we can just replace the plugin once we move to vue3? The vue3 one looks like a drop in replacement for our current one

BacLuc commented:
> Why do we want to use this plugin when we can just replace the plugin once we move to vue3? The vue3 one looks like a drop in replacement for our current one

I am not a fan of big rewrites.
If you change a lot of things, you have to test a lot of things.
Then you forget to test something, and all devs have to work with a broken main branch.
My goal ist to keep the changes for the upgrade to vue3 as small as possible, that we have a working vue3 setup with as little effort and complications as possible.
And then we can work separately to improve the different parts.

manuelmeister commented:
I vote for vite-svg-loader. It is already used in the print package and is a drop in replacement for vite-plugin-vue2-svg

BacLuc commented:
> I vote for vite-svg-loader. It is already used in the print package and is a drop in replacement for vite-plugin-vue2-svg

But not compatible with vue2

manuelmeister commented:
## Core Meeting Decision

We want to try, if the vite-svg-loader is a drop in solution, otherwise we will try this PR again.
